### ISSUE TITLE
pytest check is not reporting all failures distributed across multiple python functions or setup fixture called within a test function

### ISSUE THREAD

**Issue by C9943028** [okken/pytest-check#141]
Code has 3 functions  func A() , func B() and func C()   called a test function test_A as follows:

test_A():
func A()- 2 failures
func B()-3 failures
func C()-2 failures

All failures are not listed when it is distributed across multiple functions 
Failure in setup fixture is not listed at all

**Comment by okken** [comment]
Weird. I’ll take a look.

**Comment by okken** [comment]
I'm not able to reproduce the problem.
It looks like it's working to me.




**test_check.py**
```python
import pytest
from pytest_check import check

def a():
    check.equal(1, 1)
    check.equal(1, 2, msg='a failure 1')
    check.equal(1, 3, msg='a failure 2')

def b():
    check.equal(1, 1)
    check.equal(1, 2, msg='a failure 1, total 3')
    check.equal(1, 3, msg='b failure 2, total 4')
    check.equal(1, 4, msg='c failure 3, total 5')


def c():
    check.equal(1, 1)
    check.equal(1, 2, msg='c failure 1, total 6')
    check.equal(1, 3, msg='c failure 2, total 7')

@pytest.fixture()
def a_fixture():
    yield
    # failure in teardown
    check.equal(1, 1)
    check.equal(1, 2, msg='fixture failure 1, total 8')
    check.equal(1, 3, msg='fixture failure 2, total 9')


def test_(a_fixture):
    a()
    b()
    c()
```

It's a little confusing because the teardown failures cause an ERROR and pytest reports errors separately from failures.
Also the errors are reported first.

But still, I see 7 errors in test, and 2 in the fixture.

```
$ pytest test_check.py
============================= test session starts =============================
collected 1 item

test_check.py FE                                                         [100%]

=================================== ERRORS ====================================
_________________________ ERROR at teardown of test_ __________________________
FAILURE: check 1 == 2: fixture failure 1, total 8
test_check.py:26 in a_fixture() -> check.equal(1, 2, msg='fixture failure 1, total 8')

FAILURE: check 1 == 3: fixture failure 2, total 9
------------------------------------------------------------
Failed Checks: 2
================================== FAILURES ===================================
____________________________________ test_ ____________________________________
FAILURE: check 1 == 2: a failure 1
test_check.py:31 in test_() -> a()
test_check.py:6 in a() -> check.equal(1, 2, msg='a failure 1')

FAILURE: check 1 == 3: a failure 2
FAILURE: check 1 == 2: a failure 1, total 3
FAILURE: check 1 == 3: b failure 2, total 4
FAILURE: check 1 == 4: c failure 3, total 5
FAILURE: check 1 == 2: c failure 1, total 6
FAILURE: check 1 == 3: c failure 2, total 7
------------------------------------------------------------
Failed Checks: 7
=========================== short test summary info ===========================
FAILED test_check.py::test_
ERROR test_check.py::test_
========================= 1 failed, 1 error in 0.19s ==========================
```

**Comment by okken** [comment]
If you've got a failure example, please share, and I can re-open the issue.
